src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -5,14 +5,190 @@
+
+
+def _is_function_wellformed(ex_fn: Dict[str, Any]) -> bool:
+    tuple_type: Tuple = ()
+    if ex_fn.get("func", None):
+        if isinstance(ex_fn["func"], str):
+            return ex_fn["func"].strip() != ""
+        elif isinstance(ex_fn["func"], type(tuple_type)):
+            return isinstance(ex_fn["func"][1], str) and ex_fn["func"][1].strip() != ""
+    return False
+
+
+def sanitize_extension_function_data(ex_fn: Dict[str, Any], py_file: Path) -> bool:
+    """
+    Helper function to add in defaults for keys that do not exist in the dictionary or contain empty strings when they should not.
+    This helper function is needed because different callback sources can create dictionaries with different/missing keys, and since
+    Snowflake CLI may not own all callback implementations, the dictionaries need to have the minimum set of keys and their default
+    values to be used in creation of the SQL DDL statements.
+
+    Returns:
+    A boolean value, True if everything has been successfully validated and assigned, False if an error was encountered.
+    """
+
+    # Must have keys to create an extension function in SQL for Native Apps
+    _sanitize_ex_fn_attribute(
+        attr="object_type", ex_fn=ex_fn, make_uppercase=True, expected_type=str
+    )
+    if ex_fn["object_type"] is None:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_str(attribute="object_type", py_file=py_file)
+        )
+
+    _sanitize_ex_fn_attribute(
+        attr="object_name", ex_fn=ex_fn, make_uppercase=True, expected_type=str
+    )
+    if ex_fn["object_name"] is None:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_str(attribute="object_name", py_file=py_file)
+        )
+
+    _sanitize_ex_fn_attribute(
+        attr="return_sql", ex_fn=ex_fn, make_uppercase=True, expected_type=str
+    )
+    if ex_fn["return_sql"] is None:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_str(attribute="return_sql", py_file=py_file)
+        )
+
+    if not _is_function_wellformed(ex_fn=ex_fn):
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_str(attribute="func", py_file=py_file)
+        )
+
+    optional_expected_type: Optional[List] = []
+    default_raw_imports: List[Union[str, Tuple[str, str]]] = []
+    _sanitize_ex_fn_attribute(
+        attr="raw_imports",
+        ex_fn=ex_fn,
+        expected_type=type(optional_expected_type),
+        default_value=default_raw_imports,
+    )
+    if len(ex_fn["raw_imports"]) == 0:
+        raise MalformedExtensionFunctionError(
+            _create_missing_attr_str(attribute="raw_imports", py_file=py_file)
+        )
+
+    _sanitize_ex_fn_attribute(
+        attr="schema", ex_fn=ex_fn, make_uppercase=True, expected_type=str
+    )
+    if ex_fn["schema"] is None:
+        raise MalformedExtensionFunctionError(
```

Comment:
NB: Eventually we'll want to switch to accumulating warnings/errors instead of relying on exceptions like this. This way, we can control the output and the flow of the codegen process.

Reply:
Added TODO.